sql/analyzer/resolve_columns.go

```diff
@@ -830,6 +866,15 @@ func indexColumns(_ *sql.Context, _ *Analyzer, n sql.Node, scope *Scope) (map[ta
 		// opaque nodes have derived schemas
 		// TODO also subquery aliases?
 		indexChildNode(node.(sql.BinaryNode).Left())
+	case *plan.InsertInto:
+		// should index columns in InsertInto.Source
+		transform.Inspect(node.Source, func(n sql.Node) bool {
+			if resTbl, ok := n.(*plan.ResolvedTable); ok {
+				idx = 0
```

Comment:
Might be worth a comment on why we reset `idx` to zero here. I had to dig through the code a bit, but it looks like for an InsertInto statement, we just don't ever need the index to look up a column in the Source?

Reply:
I had to reset `idx` to 0 because this function assumes there's only one table, so the indexes would be offset for every table in `InsertInto.Source` (after indexing the columns in `InsertInto.Destination`

Reply:
Makes sense. Add that as a comment in the code so future readers don't have to reverse engineer that. 😉 The `idx` handling is a little hacky/tricky in this code in general, so it's worth explaining in the code with a comment.